python_modules/dagster/dagster/_core/definitions/repository_definition/repository_data_builder.py

```diff
@@ -229,6 +252,8 @@ def build_caching_repository_data_from_list(
             for key in definition.check_keys:
                 if key in asset_check_keys:
                     raise DagsterInvalidDefinitionError(f"Duplicate asset check key: {key}")
+            if definition.partitions_def is not None:
+                partitions_defs.add(definition.partitions_def)
```

Comment:
Ah I see we have no marker interface like `IsPartitioned` or something so the sep checks are necessary. Makes sense.